src/v3/src/transformer/terminal/transformTerminalMessages.ts

```diff
@@ -34,9 +34,10 @@ type ModifiedIdxMessage = Modify<IdxMessage, {
     key: string;
     params?: unknown[];
   };
+  title?: string;
 }>;
 
-const appendMessageElements = (uischema: UISchemaLayout, messages: IdxMessage[]): void => {
+const appendMessageElements = (uischema: UISchemaLayout, messages: ModifiedIdxMessage[]): void => {
```

Comment:
NOTE: Adding this property forces the type change on `appendMessageElements()`. Otherwise the `Modify` is just making those properties optional.

Not much to do about this unless/until we overhaul this transformer. The if/else chain below smells and should be refactored.